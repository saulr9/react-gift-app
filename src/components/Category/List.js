import { useState } from 'react';
import PropTypes from 'prop-types';
import Form from './Form';
import Grid from './Grid';
import styles from './List.module.css';

function List({ defaultCategories }) {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <Form setCategories={setCategories} />
      <div className={styles.list}>
        {categories.map((category) => (
          <Grid key={category} category={category} />
        ))}
      </div>
    </>
  );
}
List.propTypes = {
  defaultCategories: PropTypes.arrayOf(PropTypes.string)
};
List.defaultProps = {
  defaultCategories: []
};
export default List;
