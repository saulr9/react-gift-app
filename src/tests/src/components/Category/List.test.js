/* eslint-disable import/no-unresolved */
import { shallow } from 'enzyme';
import List from 'components/Category/List';

describe('Testing category List component', () => {
  let wrapper = shallow(<List />);
  const defaultCategories = ['goku', 'vegeta'];
  test('should render List component - Match SnapShot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('should there 2 grid items if there are 2 categories', () => {
    wrapper = shallow(<List defaultCategories={defaultCategories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Grid').length).toBe(defaultCategories.length);
  });
});
