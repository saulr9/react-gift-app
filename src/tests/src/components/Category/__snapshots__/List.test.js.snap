// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing category List component should render List component - Match SnapShot 1`] = `
<Fragment>
  <Form
    setCategories={[Function]}
  />
  <div
    className="list"
  />
</Fragment>
`;

exports[`Testing category List component should there 2 grid items if there are 2 categories 1`] = `
<Fragment>
  <Form
    setCategories={[Function]}
  />
  <div
    className="list"
  >
    <Grid
      category="goku"
      key="goku"
    />
    <Grid
      category="vegeta"
      key="vegeta"
    />
  </div>
</Fragment>
`;
